iib/workers/tasks/utils.py

```diff
@@ -50,6 +121,130 @@ def get_binary_image_from_config(ocp_version, distribution_scope, binary_image_c
     return binary_image
 
 
+def get_bundle_json(bundles):
+    return [json.loads(bundle) for bundle in re.split(r'(?<=})\n(?={)', bundles)]
+
+
+def _requires_max_ocp_version(bundle):
+    """
+    Check if the bundle requires the olm.maxOpenShiftVersion property.
+
+    This property is required for bundles using deprecated APIs that don't already
+    have the olm.maxOpenShiftVersion property set.
+
+    :param str bundle: a string representing the bundle pull specification
+    :returns bool:
+    """
+    cmd = [
+        'operator-sdk',
+        'bundle',
+        'validate',
+        bundle,
+        '--select-optional',
+        'name=community',
+        '--output=json-alpha1',
+        '--image-builder',
+        'none',
+    ]
+    result = run_cmd(cmd, strict=False)
+    if result:
+        output = json.loads(result)
+        # check if the bundle validation failed
+        if not output['passed']:
+            # check if the failure is due to the presence of deprecated APIs
+            # and absence of the 'olm.maxOpenShiftVersion' property
+            # Note: there is no other error in the sdk that mentions this field
+            for msg in output['outputs']:
+                if 'olm.maxOpenShiftVersion' in msg['message']:
+                    return True
+    return False
+
+
+def serve_index_registry(db_path):
```

Comment:
@yashvardhannanavati I like your idea of moving code out of build.py but utils.py is growing extremely. Can we make a new file called opm or something like that?

Reply:
A new file simply for these functions only wouldn't make sense IMO. We should file an issue and discuss it with the guild before doing a major refactoring

Reply:
sounds good :) please ping me in the jira

Reply:
Sounds good. In the meanwhile, could you please help review this PR?